src/core/execute.c

```diff
@@ -7070,6 +7085,48 @@ int exec_directory_add(ExecDirectoryItem **d, size_t *n, const char *path, char
         return 0;
 }
 
+int exec_store_log_filter_pattern(ExecContext *c, const char *pattern, bool is_allowlist) {
+        _cleanup_(pattern_freep) pcre2_code *compiled_pattern = NULL;
+        int r;
+
+        assert(c);
+        assert(pattern);
+
+        if (isempty(pattern)) {
+                /* Empty assignment resets the lists. */
+                c->log_filter_allowed_patterns = set_free(c->log_filter_allowed_patterns);
+                c->log_filter_denied_patterns = set_free(c->log_filter_denied_patterns);
+                return 0;
+        }
+
+        r = pattern_compile_and_log(pattern, 0, &compiled_pattern);
+        if (ERRNO_IS_NOT_SUPPORTED(r))
+                return r;
+        else if (r < 0)
+                return -EINVAL;
```

Comment:
Please always check if `r` is negative at first when `ERRNO_IS_XXX()` macro is used, e.g.
```
if (r < 0) {
        if (ERRNO_IS_NOT_SUPPORTED(r))
                return r;
        return -EINVAL;
}
```

BTW, I do not understand this error handling, why not to simply propagate all errors??